build_scripts/windows/scripts/build.cmd

```diff
@@ -13,28 +13,46 @@ if "%CLI_VERSION%"=="" (
     echo Please set the CLI_VERSION environment variable, e.g. 2.0.13
     goto ERROR
 )
+
+if "%ARCH%"=="" (
+    set ARCH=x86
+)
+if "%ARCH%"=="x86" (
+    set PYTHON_ARCH=win32
+) else if "%ARCH%"=="x64" (
+    set PYTHON_ARCH=amd64
+) else (
+    echo Please set ARCH to "x86" or "x64"
+    goto ERROR
+)
 set PYTHON_VERSION=3.10.10
 
 set WIX_DOWNLOAD_URL="https://azurecliprod.blob.core.windows.net/msi/wix310-binaries-mirror.zip"
-set PYTHON_DOWNLOAD_URL="https://www.python.org/ftp/python/%PYTHON_VERSION%/python-%PYTHON_VERSION%-embed-win32.zip"
+set PYTHON_DOWNLOAD_URL="https://www.python.org/ftp/python/%PYTHON_VERSION%/python-%PYTHON_VERSION%-embed-%PYTHON_ARCH%.zip"
 
 REM https://pip.pypa.io/en/stable/installation/#get-pip-py
 set GET_PIP_DOWNLOAD_URL="https://bootstrap.pypa.io/get-pip.py"
 
-REM Set up the output directory and temp. directories
-echo Cleaning previous build artifacts...
 set OUTPUT_DIR=%~dp0..\out
-if exist %OUTPUT_DIR% rmdir /s /q %OUTPUT_DIR%
-mkdir %OUTPUT_DIR%
 
-set ARTIFACTS_DIR=%~dp0..\artifacts
+REM Set up the output directory and temp. directories
+if "%NO_CLEAN%"=="" (
+    echo Cleaning previous build artifacts...
+    if exist %OUTPUT_DIR% rmdir /s /q %OUTPUT_DIR%
+    mkdir %OUTPUT_DIR%
+)
+
+set ARTIFACTS_DIR=%~dp0..\artifacts\%ARCH%
 mkdir %ARTIFACTS_DIR%
 set TEMP_SCRATCH_FOLDER=%ARTIFACTS_DIR%\cli_scratch
 set BUILDING_DIR=%ARTIFACTS_DIR%\cli
 set WIX_DIR=%ARTIFACTS_DIR%\wix
 set PYTHON_DIR=%ARTIFACTS_DIR%\Python
 
-set REPO_ROOT=%~dp0..\..\..
+REM Get the absolute directory since we pushd into different levels of subdirectories.
+PUSHD %~dp0..\..\..
+SET REPO_ROOT=%CD%
+POPD
```

Comment:
This is indeed more elegant than the current implementation. The current implementation sets `REPO_ROOT` to `D:\cli\azure-cli\build_scripts\windows\scripts\..\..\..` and the new approach sets `REPO_ROOT` to `D:\cli\azure-cli`, making it much more readable.